runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala

```diff
@@ -53,33 +53,29 @@ object AkkaHttpClientUtils {
     // Configuring compression per call could be a future power API feature.
     implicit val writer = GrpcProtocolNative.newWriter(Identity)
 
-    val target = {
-      // TODO FIXME discovery, loadbalancing etc
-      // https://github.com/akka/akka-grpc/issues/1196
-      // https://github.com/akka/akka-grpc/issues/1197
-      val resolved = settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds)
-      Await.result(resolved, 10.seconds).addresses.head
-    }
+    // TODO FIXME adapt to new API's for discovery, loadbalancing etc
+    // https://github.com/akka/akka-grpc/issues/1196
+    // https://github.com/akka/akka-grpc/issues/1197
 
-    val clientConnectionSettings = settings.overrideAuthority match {
-      case None => ClientConnectionSettings(sys)
-      case Some(authority) =>
-        ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, port) => {
+    val random = new Random()
+    val clientConnectionSettings =
+      ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, _) => {
+        settings.overrideAuthority.foreach { authority =>
           assert(host == authority)
-          assert(port == target.port.getOrElse(settings.defaultPort))
+        }
+        settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds).map { resolved =>
```

Comment:
Is that the part that fixes the fail-over?

Reply:
Basically yes: by doing the discovery inside the resolver, it is performed again for each materialization of the connection. A bit coarse perhaps, but does the trick for now here.